realm/realm-library/src/main/java/io/realm/internal/SharedRealm.java

```diff
@@ -161,10 +176,12 @@ long getNativePtr() {
 
     public void beginTransaction() {
         nativeBeginTransaction(nativePtr);
+        invokeListenerIfSchemaChanged();
     }
 
     public void commitTransaction() {
         nativeCommitTransaction(nativePtr);
+        invokeListenerIfSchemaChanged();
```

Comment:
@beeender Really? That would be rather unfortunate, because it introduces a lot of really weird situations